lib/msal-core/src/MsalTypes.ts

```diff
@@ -0,0 +1,7 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/**
+ * Key-Value type to support queryParams and extraQueryParams
+ */
+export type Dict = {[key: string]: string};
```

Comment:
I don't think `Dict` is specific enough, I would call this `StringDict` or something to make it clear values can only be strings.

Reply:
Agreed. This is something I wanted to change. Any standard naming for this "type" in JS world?

Reply:
Not sure, I'm kinda surprised Typescript doesn't have a built-in type you can use for this.

Reply:
Looks like `Record<string, string>` would work? https://www.typescriptlang.org/docs/handbook/advanced-types.html#mapped-types

Reply:
+1 if TS already has a type for this.

Reply:
Record<string, string> may be overkill. As per this SO [post]https://stackoverflow.com/questions/51936369/what-is-the-record-type-in-typescriptl) it  is better suited for not just string but any datatype and a great way to get typesafety in TS in usecases mentioned in the post. We are looking at a free flowing map<string, string>.

Naming it StringDict<>.